app/validation/validator.py

```diff
@@ -497,48 +496,65 @@ def _validate_answer_ids_present_in_schema(self, when, answer_ids_with_group_id,
+    def _validate_checkbox_exclusive_conditions_in_when_rule(self, when, answer_ids_with_group_id):
         """
-        Validate when rule with 'type' key is correct
+        Validate checkbox exclusive conditions are only used when answer type is Checkbox
         :return: list of dictionaries containing error messages, otherwise it returns an empty list
         """
         errors = []
 
         condition = when['condition']
-        checkbox_exclusive_conditions = ('contains any', 'contains all')
+        valid_checkbox_conditions = ('contains any', 'contains all', 'contains', 'not contains', 'set', 'not set')
         answer_type = answer_ids_with_group_id[when['id']]['answer']['type'] if 'id' in when else None
 
         if answer_type == 'Checkbox':
-            if condition not in checkbox_exclusive_conditions:
-                errors.append(self._error_message(f'The condition`{condition}` cannot be used with `Checkbox` answer type.'))
-        elif condition in checkbox_exclusive_conditions:
+            if condition not in valid_checkbox_conditions:
+                errors.append(self._error_message(f'The condition `{condition}` cannot be used with `Checkbox` answer type.'))
+        elif condition in ('contains any', 'contains all', 'contains', 'not contains'):
             errors.append(
-                self._error_message(f'The condition`{condition}` can only be used with `Checkbox` answer types. Found answer type: {answer_type}'))
+                self._error_message(f'The condition `{condition}` can only be used with `Checkbox` answer types. Found answer type: {answer_type}'))
+
+        return errors
 
+    def _validate_conditions_requiring_list_match_values_are_valid_in_when_rule(self, when, answer_ids_with_group_id, referenced_id):
+        """
+        Validate conditions requiring list match values define the values or comparison id property
+        :return: list of dictionaries containing error messages, otherwise it returns an empty list
+        """
+        errors = []
+        condition, answer_id = when['condition'], when['id']
         conditions_requiring_list_match_values = ('equals any', 'not equals any', 'contains any', 'contains all')
-        if condition in conditions_requiring_list_match_values and not when.get('values'):
-            errors.append(self._error_message(f'Property `values` is missing/empty for when condition `{condition}` with answer id `{when["id"]}`'))
+
+        if condition in conditions_requiring_list_match_values and not when.get('values', when.get('comparison_id')):
+            errors.append(self._error_message(
+                f'Either the properties `values` or `comparison_id` is missing/empty for when condition `{condition}` with answer id `{answer_id}`'))
+
+        if 'comparison_id' in when:
+            errors.extend(
+                self._validate_comparison_id_in_when_rule(when, answer_ids_with_group_id, referenced_id, conditions_requiring_list_match_values))
```

Comment:
It's not clear that this method is also validating comparison_id for non list match values as it's named `_validate_conditions_requiring_list_match_values_are_valid_in_when_rule`.

It would be better to call `_validate_comparison_id_in_when_rule` first and from there call this method, or just move that logic into that method.

Reply:
Have removed `_validate_conditions_requiring_list_match_values_are_valid_in_when_rule` as JSON schema takes care of it anyway and kept `_validate_comparison_id_in_when_rule`.